chapters/package-information.md

```diff
@@ -295,7 +295,7 @@ The metadata for the package download location field is shown in Table 19.
 | --------- | ----- |
 | Required | Yes |
 | Cardinality | 1..1 |
-| Format | Uniform resource locator \| VCS location \| `NONE` \| `NOASSERTION`<br><br>For version-controlled files, a VCS location is similar to a URL and has the following syntax:<br><br>`<vcs_tool>+<transport>://<host_name>[/<path_to_repository>][@<revision_tag_or_branch>][#<sub_path>]`<br><br>This VCS location compact notation (inspired and mostly adopted from [pip][pip-vcs] as of 2015-02-20) supports referencing locations in version control systems such as [Git][], [Mercurial][], [Subversion][] and [Bazaar][], and specifies the type of VCS tool using url prefixes: `git+`, `hg+`, `bzr+`, `svn+` and specific transport schemes such as SSH or HTTPS.<br><br>Specifying sub-paths, branch names, a commit hash, a revision or a tag name is recommended, and supported using the `@` delimiter for commits and the `#` delimiter for sub-paths.<br><br>Using user names and password in the `<host_name>` is not supported and should be considered as an error. User access control to URLs or VCS repositories shall be handled outside of an SPDX document.<br><br>In VCS location compact notations, the trailing slashes in `<host_name>`, `<path_to_repository>` are not significant. Leading and trailing slashes in `<sub_path>` are not significant. |
+| Format | Uniform resource locator \| VCS location \| `NONE` \| `NOASSERTION`<br><br>For version-controlled files, a VCS location is similar to a URL and has the following syntax:<br><br>`<vcs_tool>+<transport>://<host_name>[/<path_to_repository>][@<revision_tag_or_branch>][#<sub_path>]`,<br><br>where `@<revision_tag_or_branch>` is either a git tag or a git branch. It is highly recommended to use a full path like `refs/heads/<branch>` or `refs/tags/<tag>`, rather than only `<branch>` or `<tag>`, in order to prevent issues caused by shadowing (see [gitrevisions](https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitrevisions.html) manual page). <br><br>This VCS location compact notation (inspired and mostly adopted from [pip][pip-vcs] as of 2015-02-20) supports referencing locations in version control systems such as [Git][], [Mercurial][], [Subversion][] and [Bazaar][], and specifies the type of VCS tool using url prefixes: `git+`, `hg+`, `bzr+`, `svn+` and specific transport schemes such as SSH or HTTPS.<br><br>Specifying sub-paths, branch names, a commit hash, a revision or a tag name is recommended, and supported using the `@` delimiter for commits and the `#` delimiter for sub-paths.<br><br>Using user names and password in the `<host_name>` is not supported and should be considered as an error. User access control to URLs or VCS repositories shall be handled outside of an SPDX document.<br><br>In VCS location compact notations, the trailing slashes in `<host_name>`, `<path_to_repository>` are not significant. Leading and trailing slashes in `<sub_path>` are not significant. |
```

Comment:
Q: Should this allow any other form of `<rev>`? I believe the following are the only ones that make sense:

* `<describeOutput>`, `<rev>~[<n>]`, and/or `<rev>^[<n>]` would allow someone to symbolically reference a commit that does not have a tag or branch pointing directly to it. On the one hand, that might be valuable in some cases, since it is more descriptive than a raw commit. On the other, it might be better to push people toward simplicity.

* `<rev>:<path>` would allow referencing a particular tree or blob within the repository, rather than the full commit. I don't know if this is desirable or not. (We'd have to disallow `<path>` starting with `./` or `../` since those are relative to the current working directory, which is undefined.)

* `<rev>^{<type>}` / `<ref>^{}` would allow referencing a particular object type. This *might* be useful if one wants to say that only the tree mattered, not the full commit, though I'm not sure anyone would ever use this in practice.

I suspect we do NOT want to allow the following:

* `...@...`, `:[<n>:]<path>`: relative to the local copy of the repository (HEAD, reflog, branch configuration, or index)
* `<rev>^{/<text>}`: too complex; can be described by one of the supported values instead.
* `:/<text>`: both problems combined!

Reply:
Regarding `<describeOutput>`, `<rev>~[<n>]`, and/or `<rev>^[<n>]`, I think we would definitely like to allow, given that: 
* contents of a branch might change over time
* in case the build is cut-off on such a reference, it's better if the sbom reflects that

`<rev>:<path>` - not sure if this should be encouraged. Do you know any good example? The only reason to allow that IMO is if this is anyway how the build is created and we want that information to be visible. Which in other hand might be a good enough reason to allow?
Same with `<rev>^{<type>}` / `<ref>^{}` - I cannot think of a practical use-case to reference a build to that, but if you know any example, this might make sense.

Absolutely agree on the last 3.